lib/Analysis/DxilSimplify.cpp

```diff
@@ -72,7 +75,9 @@ bool CanSimplify(const llvm::Function *F) {
 ///
 /// If this call could not be simplified returns null.
 Value *SimplifyDxilCall(llvm::Function *F, ArrayRef<Value *> Args,
-                        llvm::Instruction *I) {
+                        const llvm::Instruction *I,
+                        llvm::Instruction *MutableI)
```

Comment:
The way this is presently used by callers of this amounts to a boolean. Do you foresee a scanario where `MutableI` might be non-null, but also not equal `I`?

Reply:
I agree this is a confusing pattern.  Might it be preferable to have a bool that indicates whether it's ok to modify the function containing `I`?  Perhaps `bool allowChanges`?

Reply:
Some callers have only a const Instruction, so the function can't just have one non-const instruction parameter. I changed it to two separate functions: `SimplifyDxilCall` and `SimplifyDxilCallMayInsert`. The first one takes a const instruction and never inserts, the second takes just a non-const instruction and is allowed to insert.